src/main/java/com/google/devtools/build/lib/runtime/BlazeRuntime.java

```diff
@@ -593,23 +594,18 @@ public static void main(Iterable<Class<? extends BlazeModule>> moduleClasses, St
    * @return the filtered request to write to the log.
    */
   @VisibleForTesting
-  public static String getRequestLogString(List<String> requestStrings) {
+  static String getRequestLogString(List<String> requestStrings) {
     StringBuilder buf = new StringBuilder();
     buf.append('[');
     String sep = "";
+    Matcher m = suppressFromLog.matcher("");
     for (String s : requestStrings) {
       buf.append(sep);
-      if (s.startsWith("--client_env")) {
-        int varStart = "--client_env=".length();
-        int varEnd = s.indexOf('=', varStart);
-        String varName = s.substring(varStart, varEnd);
-        if (suppressFromLog.matcher(varName).find()) {
-          buf.append("--client_env=");
-          buf.append(varName);
-          buf.append("=__private_value_removed__");
-        } else {
-          buf.append(s);
-        }
+      m.reset(s);
+      if (m.lookingAt()) {
+        buf.append("--client_env=");
+        buf.append(m.group(1));
```

Comment:
You could just use `m.group()` here, which would be equivalent to `--client_env=m.group(1)=`. So this code could be simplified to

```
buf.append(m.group()).append("__private_value_removed__").
```